packages/api/lambdas/sf-event-sqs-to-db-records.js

```diff
@@ -82,50 +119,124 @@ const shouldWriteExecutionToRDS = async (cumulusMessage, knex) => {
-  }
+  });
 
-  try {
-    return await knex.transaction(async (trx) => {
-      await trx(tableNames.executions)
-        .insert({
-          arn: executionArn,
-          cumulus_version: getMessageCumulusVersion(cumulusMessage),
-        });
-      return executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
+const savePdrViaTransaction = async ({
+  cumulusMessage,
+  collection,
+  provider,
+  trx,
+}) =>
+  trx(tableNames.pdrs)
+    .insert({
+      name: getMessagePdrName(cumulusMessage),
+      status: getWorkflowStatus(cumulusMessage),
+      collectionCumulusId: collection.cumulusId,
+      providerCumulusId: provider.cumulusId,
     });
-  } catch (error) {
-    log.error(`Failed to write execution records for ${executionArn}`, error);
-    throw error;
+
+const savePdr = async ({
+  cumulusMessage,
+  collection,
+  provider,
+  knex,
+  pdrModel = new Pdr(),
+}) => {
+  // If there is no PDR in the message, then there's nothing to do here, which is fine
+  if (!messageHasPdr(cumulusMessage)) {
+    return true;
+  }
+  if (!isRecordDefined(collection)) {
+    throw new Error(`Collection reference is required for a PDR, got ${collection}`);
   }
+  if (!isRecordDefined(provider)) {
+    throw new Error(`Provider reference is required for a PDR, got ${provider}`);
+  }
+  return knex.transaction(async (trx) => {
+    await savePdrViaTransaction({ cumulusMessage, collection, provider, trx });
+    return pdrModel.storePdrFromCumulusMessage(cumulusMessage);
+  });
 };
 
-const savePdrToDb = async (cumulusMessage) => {
-  const pdrModel = new Pdr();
+const saveGranulesToDb = async (cumulusMessage) => {
+  const granuleModel = new Granule();
+
   try {
-    await pdrModel.storePdrFromCumulusMessage(cumulusMessage);
+    await granuleModel.storeGranulesFromCumulusMessage(cumulusMessage);
   } catch (error) {
     const executionArn = getMessageExecutionArn(cumulusMessage);
-    log.fatal(`Failed to create/update PDR database record for execution ${executionArn}: ${error.message}`);
+    log.fatal(`Failed to create/update granule records for execution ${executionArn}: ${error.message}`);
     throw error;
   }
 };
 
-const saveGranulesToDb = async (cumulusMessage) => {
+const saveRecordsToDynamoDb = async (cumulusMessage) => {
+  const executionModel = new Execution();
+  const pdrModel = new Pdr();
   const granuleModel = new Granule();
 
+  const results = await Promise.allSettled([
+    executionModel.storeExecutionFromCumulusMessage(cumulusMessage),
+    pdrModel.storePdrFromCumulusMessage(cumulusMessage),
+    granuleModel.storeGranulesFromCumulusMessage(cumulusMessage),
+  ]);
+  const failures = results.filter((result) => result.status === 'rejected');
+  if (failures.length > 0) {
+    const allFailures = failures.map((failure) => failure.reason);
+    log.error(allFailures.join(' '));
+    throw new Error('Failed writing some records to Dynamo');
```

Comment:
We could probably make this a bit more descriptive though, given we have `allFailures`

Reply:
Also what does the record in the DLQ look like for this failure as compared to the implementation on main? 

*edit - answered below*